scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Handshaker.java

```diff
@@ -1118,4 +1125,28 @@ public void verifyCertificate(CertificateMessage message) throws HandshakeExcept
 			}
 		}
 	}
+
+	/**
+	 * Amends the peer principal with additional application level information.
+	 */
+	private void amendPeerPrincipal() {
+
+		Principal peerIdentity = session.getPeerIdentity();
+		if (peerIdentity instanceof ExtensiblePrincipal) {
+			// amend the client principal with additional application level information
+			@SuppressWarnings("unchecked")
+			ExtensiblePrincipal<? extends Principal> extensibleClientIdentity = (ExtensiblePrincipal<? extends Principal>) peerIdentity;
```

Comment:
// amend the **peer** principal with additional application level information
@SuppressWarnings("unchecked")
ExtensiblePrincipal<? extends Principal> **extensibleIdentity** = (ExtensiblePrincipal<? extends Principal>) peerIdentity;